pkg/database/vercode.go

```diff
@@ -97,11 +98,54 @@ func (v *VerificationCode) AfterCreate(scope *gorm.Scope) {
 				SET codes_issued = authorized_app_stats.codes_issued + 1
 		`
 
-		day := time.Now().UTC().Truncate(24 * time.Hour)
 		if err := scope.DB().Exec(sql, day, v.IssuingAppID).Error; err != nil {
 			scope.Log(fmt.Sprintf("failed to update stats: %v", err))
 		}
 	}
+
+	// Update the per-realm stats.
+	sql := `
+		INSERT INTO realm_stats(date, realm_id, codes_issued)
+			VALUES ($1, $2, 1)
+		ON CONFLICT (date, realm_id) DO UPDATE
+			SET codes_issued = realm_stats.codes_issued + 1
+	`
+
+	if err := scope.DB().Exec(sql, day, v.RealmID).Error; err != nil {
+		scope.Log(fmt.Sprintf("failed to update stats: %v", err))
+	}
+}
+
+// BeforeSave is called after the verification code is saved to the DB.
+func (v *VerificationCode) BeforeSave(scope *gorm.Scope) error {
```

Comment:
Hmm - doing this in BeforeSave puts us in a spot where we'd increment the stats, but then some validation could prevent the code from actually being claimed.

Reply:
Ok, I'm going to a side value in VerificationCode. This is too clunky.

Reply:
Okay - what are your thoughts on making a `Claim` function on the verification code (or on the database) that handles all this logic?

Reply:
LMK what you think of this. It's clunky, but with Hooks (and ORM with dirty) this feels the best we can do. I'm happy to move to a Claim API if you think that's best.

Reply:
Moved to claim. LMK.